### PR TITLE
add astyle, codelldb, netcoredbg,

### PR DESCRIPTION
**Pull request by danielnachun**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by danielnachun** [comment]
@conda-forge/staged-recipes This is now ready for review.

I had to disable the Windows build for all of these packages for various reasons:
- `astyle` - it needs C++17 support but even if I set the compiler to `vs2022` it does not seem to actually enable support.  I will look into this more once the feedstock is made
- `codelldb` - it cannot find `liblldb.lib` even though I explicitly specified the path.  I will also look into this more when the feedstock is made.
- `netcoredbg` - it fails due to a file path that is too long.  I don't know how this can easily be resolved but it is a recurring issue for building some packages on Windows.

**Comment by ocefpaf** [comment]
> I had to disable the Windows build for all of these packages for various reasons:

Open an issue in the feedstock, when it gets created, with those notes to make them easier to find.

File: recipes/astyle/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+set -o xtrace -o nounset -o pipefail -o errexit
+
+cmake -S . -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_VERBOSE_MAKEFILE=ON \
+    -Wno-dev \
+    -DBUILD_TESTING=OFF \
+    ${CMAKE_ARGS}
+cmake --build build -j${CPU_COUNT}
+cmake --install build
```
Comment (ocefpaf) [inline recipes/astyle/build.sh:12]: Is there a `make check` or `ctest` that we can run?

File: recipes/astyle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - cmake
+    - make
+
+test:
+  commands:
+    - {{ name }} -h
```
Comment (ocefpaf) [inline recipes/astyle/meta.yaml:26]: The excess of jinja makes the recipe hard to read and doesn't really save us anything.

```suggestion
    - astyle -h
```

File: recipes/astyle/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{% set name = "astyle" %}
+{% set version = "3.5" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://downloads.sourceforge.net/project/astyle/astyle/astyle%203.5/astyle-{{ version }}.tar.bz2
```
Comment (ocefpaf) [inline recipes/astyle/meta.yaml:9]: This is required for the bot to work.

```suggestion
  url: https://downloads.sourceforge.net/project/astyle/astyle/astyle%20{{ version }}/astyle-{{ version }}.tar.bz2
```

File: recipes/codelldb/build.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+cmake -S . -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_VERBOSE_MAKEFILE=ON \
+    -Wno-dev \
+    -DBUILD_TESTING=OFF \
+    -DLLDB_PACKAGE=${BUILD_PREFIX} \
+    ${CMAKE_ARGS}
+
+cmake --build build -j${CPU_COUNT}
+
```
Comment (ocefpaf) [inline recipes/codelldb/build.sh:20]: Any `ctest` or make `check`?

File: recipes/codelldb/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - cargo-bundle-licenses
+    - cmake
+    - lldb
+    - libcxx
+    - nodejs
+    - make
+
+test:
+  commands:
+    - {{ name }} --help
```
Comment (ocefpaf) [inline recipes/codelldb/meta.yaml:31]: Please remove the extra jinja.

File: recipes/netcoredbg/build.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+cmake -S . -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_VERBOSE_MAKEFILE=ON \
+    -Wno-dev \
+    -DBUILD_TESTING=OFF \
+    -DDOTNET_DIR=${DOTNET_ROOT} \
+    ${CMAKE_ARGS}
+cmake --build build -j${CPU_COUNT}
+cmake --install build
```
Comment (ocefpaf) [inline recipes/netcoredbg/build.sh:17]: Any test we can run here?

File: recipes/astyle/build.sh (hunk 1 above)
Comment (danielnachun) [reply recipes/astyle/build.sh:12]: Unfortunately there is no testing functionality implemented in the `CMakeLists.txt` for this package.

File: recipes/codelldb/build.sh (hunk 4 above)
Comment (danielnachun) [reply recipes/codelldb/build.sh:20]: While the `CMakeLists.txt` for this package does have a couple of test targets, all of them appear to require quite a complex setup that would be very difficult to set up in CI.

File: recipes/astyle/meta.yaml (hunk 3 above)
Comment (danielnachun) [reply recipes/astyle/meta.yaml]: I don't know how I missed this before, but this project also has source archive releases on GitLab with URLs that should be parseable by the autotick bot, so I've changed the URL over to there instead.

**Comment by danielnachun** [comment]
> > I had to disable the Windows build for all of these packages for various reasons:
> 
> Open an issue in the feedstock, when it gets created, with those notes to make them easier to find.

This is a great idea, and I will go back to the handful of other native feedstocks I maintain that don't currently build on Windows or non-Intel architectures and make issues for those as well.

File: recipes/codelldb/build.sh (hunk 4 above)
Comment (ocefpaf) [reply recipes/codelldb/build.sh:20]: Can you elaborate? How complex?

File: recipes/codelldb/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    - libcxx
+    - nodejs
+    - make
+
+test:
+  commands:
+    - codelldb --help
+
+about:
+  home: https://github.com/vadimcn/{{ name }}
```
Comment (ocefpaf) [inline recipes/codelldb/meta.yaml:34]: Please remove all the excess use of jinja. Like this the URL is not clickable. This kind of jinja doesn't help and obscure the recipe.

File: recipes/netcoredbg/build.sh (hunk 6 above)
Comment (ocefpaf) [reply recipes/netcoredbg/build.sh:17]: Same questions as above. Can you elaborate? How complex?

File: recipes/netcoredbg/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+test:
+  commands:
+    - netcoredbg --help
+
+about:
+  home: https://github.com/Samsung/{{ name }}
+  license: MIT
+  license_file: LICENSE
+  summary: NetCoreDbg is a managed code debugger with MI interface for CoreCLR.
+  dev_url: https://github.com/Samsung/{{ name }}
```
Comment (ocefpaf) [inline recipes/netcoredbg/meta.yaml:37]: Same request to remove all unnecessary jinja.

**Comment by danielnachun** [comment]
Closing for now until I have time to reorganize this.

**Pull request closed without merging**